generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/filter/LoggerRequestDataFilter.java

```diff
@@ -0,0 +1,169 @@
+        Optional.ofNullable(headers.getFirst(ACCEPT)).ifPresent(value -> builder.add(ACCEPT, value));
+
+        mergeHeadersWithPayloadMetadataIfPresent(requestContext, headers)
+                .ifPresent(metadataBuilder -> builder.add(METADATA, metadataBuilder));
+
+        MDC.put(REQUEST_DATA, builder.build().toString());
+
+        trace(logger, () -> "Request data added to MDC");
+    }
+
+    @Override
+    public void filter(final ContainerRequestContext requestContext, final ContainerResponseContext responseContext) throws IOException {
+        trace(logger, () -> "Clearing MDC");
+        MDC.clear();
+    }
+
+    private void addServiceContextNameIfPresent(final JsonObjectBuilder builder) {
+        Optional.ofNullable(serviceContextNameProvider.getServiceContextName())
+                .ifPresent(value -> builder.add(SERVICE_CONTEXT, value));
+    }
+
+    private Optional<JsonObject> mergeHeadersWithPayloadMetadataIfPresent(final ContainerRequestContext requestContext,
+                                                                          final MultivaluedMap<String, String> headers) throws IOException {
+        Optional<String> id = Optional.ofNullable(headers.getFirst(ID));
+        Optional<String> name = Optional.ofNullable(headers.getFirst(NAME));
+        Optional<String> correlationId = Optional.ofNullable(headers.getFirst(CLIENT_CORRELATION_ID));
+        Optional<String> sessionId = Optional.ofNullable(headers.getFirst(SESSION_ID));
+        Optional<String> userId = Optional.ofNullable(headers.getFirst(USER_ID));
+
+        final Optional<MediaType> mediaType = Optional.ofNullable(requestContext.getMediaType());
+
+        if (isPayloadPresent(mediaType)) {
+            final String charset = extractCharset(mediaType.get());
+            final String payload = IOUtils.toString(requestContext.getEntityStream(), charset);
+
+            if (payload.contains(PAYLOAD_METADATA)) {
+                final Metadata metadata = payloadMetadataFrom(payload);
+
+                if (!id.isPresent()) {
+                    id = Optional.of(metadata.id().toString());
+                }
+
+                if (!name.isPresent()) {
+                    name = Optional.of(metadata.name());
+                }
+
+                if (!correlationId.isPresent()) {
+                    correlationId = metadata.clientCorrelationId();
+                }
+
+                if (!sessionId.isPresent()) {
+                    sessionId = metadata.sessionId();
+                }
+
+                if (!userId.isPresent()) {
+                    userId = metadata.userId();
+                }
+            }
+
+            requestContext.setEntityStream(new ByteArrayInputStream(payload.getBytes(charset)));
+        }
+
+        if (id.isPresent() || name.isPresent() || correlationId.isPresent() || sessionId.isPresent() || userId.isPresent()) {
```

Comment:
The reason for this was to remove completely the metadata object if it was not required.